recipes/jasper/all/conanfile.py

```diff
@@ -29,92 +28,76 @@ class JasperConan(ConanFile):
         if self.options.with_libjpeg == "libjpeg-turbo":
-            self.requires("libjpeg-turbo/2.1.2")
+            self.requires("libjpeg-turbo/2.1.4")
         elif self.options.with_libjpeg == "libjpeg":
             self.requires("libjpeg/9e")
 
-    def validate(self):
-        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) == "16":
-            raise ConanInvalidConfiguration(f"{self.name} Current can not build in CCI due to windows SDK version. See https://github.com/conan-io/conan-center-index/pull/13285 for the solution hopefully")
-
     def source(self):
         get(self, **self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["JAS_ENABLE_DOC"] = False
-        self._cmake.definitions["JAS_ENABLE_PROGRAMS"] = False
-        self._cmake.definitions["JAS_ENABLE_SHARED"] = self.options.shared
-        self._cmake.definitions["JAS_LIBJPEG_REQUIRED"] = "REQUIRED"
-        self._cmake.definitions["JAS_ENABLE_OPENGL"] = False
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
-    def _patch_sources(self):
-        apply_conandata_patches(self)
-        # Clean rpath in installed shared lib
-        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
-        cmds_to_remove = [
-            "set(CMAKE_INSTALL_RPATH \"${CMAKE_INSTALL_PREFIX}/lib\")",
-            "set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)",
-            "set(CMAKE_INSTALL_RPATH\n		  \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}\")",
-        ]
-        for cmd_to_remove in cmds_to_remove:
-            replace_in_file(self, cmakelists, cmd_to_remove, "")
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["JAS_ENABLE_DOC"] = False
+        tc.variables["JAS_ENABLE_LATEX"] = False
+        tc.variables["JAS_ENABLE_PROGRAMS"] = False
+        tc.variables["JAS_ENABLE_SHARED"] = self.options.shared
+        tc.variables["JAS_LIBJPEG_REQUIRED"] = "REQUIRED"
+        tc.variables["JAS_ENABLE_OPENGL"] = False
+        tc.variables["JAS_ENABLE_LIBJPEG"] = True
+        if cross_building(self):
+            tc.cache_variables["JAS_CROSSCOMPILING"] = True
+            tc.cache_variables["JAS_STDC_VERSION"] = "199901L"
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+
+        # TODO: Remove after fixing https://github.com/conan-io/conan-center-index/issues/13159
+        # C3I workaround to force CMake to choose the highest version of
+        # the windows SDK available in the system
+        if is_msvc(self) and not self.conf.get("tools.cmake.cmaketoolchain:system_version"):
+            tc.variables["CMAKE_SYSTEM_VERSION"] = "10.0"
```

Comment:
@uilianries  this needs to be passed to `CMakeToolchain`, whole block needs to be moved a few lines above :P

Reply:
Fixed! Thank you for pointing it!